.github/workflows/npm-publish.yml

```
@@ -30,21 +30,4 @@ jobs:
       - run: npm ci
       - run: npm publish
```

```suggestion
      - run: npm ci
      - run: npm test
      - run: npm build
      - run: npm publish
```

Then delete the needs on line 22 and the whole build step.  Splitting this into two jobs doesn't seem beneficial in this case

---

I'm assuming we need a build step and publish doesn't handle that, but I could be wrong - you know the code best